master/buildbot/steps/artifact.py

```diff
@@ -367,8 +385,13 @@ def start(self):
         triggeredbybrid = self.build.requests[0].id
         br = yield self.master.db.buildrequests.getBuildRequestTriggered(triggeredbybrid, self.artifactBuilderName)
 
-        artifactPath  = "%s_%s_%s" % (safeTranslate(self.artifactBuilderName),
-                                      br['brid'], FormatDatetime(br["submitted_at"]))
+        if br["submitted_at"] > ARTIFACT_LOCATION_CHANGE_DATE:
+            artifactPath  = "%s/%s_%s" % (safeTranslate(self.artifactBuilderName),
```

Comment:
Other places use `/`:

```
def getRemoteLocation(artifactServer, artifactServerDir, artifactPath, artifact):
    return artifactServer + ":" + artifactServerDir + "/" + artifactPath + "/" + artifact.replace(" ", r"\ ")
```

Since we are always calling `rsync`, I think that using `/` is the safest way. `os.path.join` in Windows would join paths with `\`, and IDK if that works for `rsync` (and if Katana doesn't work in windows, no big reason to use `os.path` anyway).